### ISSUE TITLE
Add OSGi instructions to bundle

### ISSUE THREAD

**Issue by smigfu** [msgpack/msgpack-java#352]
Could we add OSGi headers to the MANIFEST.MF so the project can be used in an OSGi environment?

**Comment by xerial** [comment]
Could you check #353 and generated bundle? https://oss.sonatype.org/content/repositories/snapshots/org/msgpack/msgpack-core/0.8.4-SNAPSHOT/

I'm not sure the exported package is sufficient for OSGi.

**Comment by smigfu** [comment]
Hey Taro, thanks for looking into this. I am sorry I didn't provide a PR with the headers myself, but I am not too familiar with sbt. Anyhow I have checked the generated bundles and there are some imports missing. I have attached the MANIFEST.MFs for msgpack.core and jackson-dataformat-msgpack as they should look like (I have generated those by using the OSGi-BND tools).

`Manifest-Version: 1.0
Bundle-Description: Shaded and OSGi-fied
Bundle-SymbolicName: org.msgpack.msgpack-core
Built-By: pmarx
Bundle-ManifestVersion: 2
Bnd-LastModified: 1457364957426
Import-Package: org.msgpack.core,org.msgpack.core.annotations,org.msgp
 ack.core.buffer,org.msgpack.value,org.msgpack.value.impl,sun.misc
Tool: Bnd-0.0.357
Export-Package: org.msgpack.value;uses:="org.msgpack.core,org.msgpack.
 value.impl",org.msgpack.value.impl;uses:="org.msgpack.value,org.msgpa
 ck.core",org.msgpack.core;uses:="org.msgpack.core.buffer,org.msgpack.
 value",org.msgpack.core.annotations,org.msgpack.core.buffer;uses:="or
 g.msgpack.core,sun.misc"
Bundle-Name: msgpack-core
Bundle-Version: 0.8.3.OSGI
Created-By: Apache Maven Bundle Plugin
Build-Jdk: 1.8.0_40
`

`Manifest-Version: 1.0
Bundle-Description: Shaded and OSGi-fied
Bundle-SymbolicName: org.msgpack.jackson-dataformat-msgpack
Built-By: pmarx
Bundle-ManifestVersion: 2
Bnd-LastModified: 1457364425293
Import-Package: com.fasterxml.jackson.core;version="[2.7,3)",com.faste
 rxml.jackson.core.base;version="[2.7,3)",com.fasterxml.jackson.core.i
 o;version="[2.7,3)",com.fasterxml.jackson.core.json;version="[2.7,3)"
 ,com.fasterxml.jackson.databind;version="[2.7,3)",com.fasterxml.jacks
 on.databind.annotation;version="[2.7,3)",org.msgpack.core,org.msgpack
 .core.buffer,org.msgpack.jackson.dataformat,org.msgpack.value
Tool: Bnd-0.0.357
Export-Package: org.msgpack.jackson.dataformat;uses:="com.fasterxml.ja
 ckson.core,com.fasterxml.jackson.databind,com.fasterxml.jackson.datab
 ind.annotation,com.fasterxml.jackson.core.io,com.fasterxml.jackson.co
 re.json,org.msgpack.core.buffer,org.msgpack.core,com.fasterxml.jackso
 n.core.base,org.msgpack.value"
Bundle-Name: jackson-dataformat-msgpack
Bundle-Version: 0.8.3.OSGI
Created-By: Apache Maven Bundle Plugin
Build-Jdk: 1.8.0_40`

**Comment by smigfu** [comment]
[MANIFEST.MF.TXT](https://github.com/msgpack/msgpack-java/files/166982/MANIFEST.MF.TXT)
[MANIFEST.MF.TXT](https://github.com/msgpack/msgpack-java/files/166983/MANIFEST.MF.TXT)

**Comment by xerial** [comment]
Thanks for the details. Yes. This manifest is useful to modify the sbt settings:
```Bundle-Description: Shaded and OSGi-fied
Bundle-SymbolicName: org.msgpack.msgpack-core
Built-By: pmarx
Bundle-ManifestVersion: 2
Bnd-LastModified: 1457364957426
Import-Package: org.msgpack.core,org.msgpack.core.annotations,org.msgp
 ack.core.buffer,org.msgpack.value,org.msgpack.value.impl,sun.misc
Tool: Bnd-0.0.357
Export-Package: org.msgpack.value;uses:="org.msgpack.core,org.msgpack.
 value.impl",org.msgpack.value.impl;uses:="org.msgpack.value,org.msgpa
 ck.core",org.msgpack.core;uses:="org.msgpack.core.buffer,org.msgpack.
 value",org.msgpack.core.annotations,org.msgpack.core.buffer;uses:="or
 g.msgpack.core,sun.misc"
Bundle-Name: msgpack-core```

**Comment by mattbishop** [comment]
FWIW 0.6.x had OSGi manifests. This is a regression.

**Comment by mattbishop** [comment]
#353 looks OK, though you don't need to import-package the osgi framework AFAIK.

**Comment by xerial** [comment]
Released 0.8.4 with the OSGi bundle properties.
